quickwit/quickwit-search/src/collector.rs

```diff
@@ -548,46 +550,27 @@ fn map_error(err: postcard::Error) -> TantivyError {
 }
 
 /// Merges a set of Leaf Results.
-fn merge_leaf_responses(
+fn merge_intermediate_aggregation_result<'a>(
     aggregations_opt: &Option<QuickwitAggregations>,
-    mut leaf_responses: Vec<LeafSearchResponse>,
-    sort_order1: SortOrder,
-    sort_order2: SortOrder,
-    max_hits: usize,
-) -> tantivy::Result<LeafSearchResponse> {
-    // Optimization: No merging needed if there is only one result.
-    if leaf_responses.len() == 1 {
```

Comment:
It's a bummer we lose this optimization.

Reply:
We don't lose it, it's at the start of `fn merge_leaf_responses()`, which call this function. In a previous iteration it was called from 2 places, but right now it's only called from one, so it could re-inlined there actually

Reply:
Ah awesome!